### DISCUSSION TITLE
ESP32-S3 microcontrollers - BNN

### DISCUSSION THREAD

**Discussion by christophezeinaty** [larq/compute-engine#790] [General]
Hello I have a question concerning an info I read  [here](https://plumerai.com/people-detection-live), it mention that the tiny model for person detection is tested on ESP32-S3 microcontrollers but if I understood correctly the larq compute engine optimizations only works on arm based accelerators so how did you test that on ESP?

**Comment by CNugteren** [comment]
The Larq Compute Engine is indeed targeted at Cortex-A class Arm CPUs, and is meant to accelerate BNN models.

What you've seen on the Plumerai website is one of our trained AI models that we offer commercially. In that case we do include an inference engine that runs on smaller processors, such as the ESP32-S3 indeed. See for more information [our docs](https://docs.plumerai.com/latest/people_detection_micro/#index) or [our blog](https://blog.plumerai.com/2023/03/espressif_esp32-s3_support/). However, this is not available in free or open-source form.

**Reply by christophezeinaty** [reply]
ok thank you for the clarification, so I can assume as well that for 32 bits Arm CPUs, BNN optimization won't work with the Larq Compute Engine?

**Reply by CNugteren** [reply]
It is possible to build the Larq Compute Engine for 32-bit Arm CPUs / OS-es, see [the docs](https://docs.larq.dev/compute-engine/build/arm/). However, note that this still means Cortex-A class Arm CPUs (e.g. armv7 architecture), and not e.g. Cortex-M.

**Reply by christophezeinaty** [reply]
Ah ok well noted thank you for your reactivity !
